### ISSUE TITLE
Generated client is not compatible when typescript has noUnusedParameters set to true

### ISSUE THREAD

**Issue by Nysosis** [acacode/swagger-typescript-api#85]
When using this library with my project, that has the tsconfig setting of `noUnusedParameters` set to true, I get errors with the generated client at `safeParseResponse`

This is because neither the `E` generic type is used, nor is the captured `e` variable in the catch handler (see attached)

![image](https://user-images.githubusercontent.com/5476382/96711706-e5b79280-1395-11eb-9a36-3477f8b74f57.png)

I resolve this locally by prefixing both variables with an underscore, but it would be nice not to have to re-apply this every time I regenerate the client.

Looking at the [client template](https://github.com/acacode/swagger-typescript-api/blob/a163999a46a40d99fc73027684e9d0c8bac6ca9f/src/templates/client.mustache#L96) it appears the `E` generic type is sometimes used, when the config setting `generateResponses` is specified (which I'm not currently), but not in all instances, so fix will need to respect that.

But the caught `e` variable is never used, so that could simply be removed?

**Comment by js2me** [comment]
Hello @Nysosis !   
Thanks for this issue, I'll fix it at the next major release.

**Comment by js2me** [comment]
> But the caught e variable is never used, so that could simply be removed?  
@Nysosis I think `e` should be return from safeParseResponse, what do you think about that?  

The current version on the next branch   
```ts

interface ClientResponse<D extends unknown, E extends unknown = unknown> extends Response {
  data: D | null;
  error: E | null;
}

//

  private safeParseResponse = <T = any, E = any>(response: Response): Promise<ClientResponse<T, E>> => {
    const r = response.clone() as ClientResponse<T, E>;
    r.data = null;
    r.error = null;

    return response
      .json()
      .then((data) => {
        r.data = data;
        return r;
      })
      .catch((e) => {
        r.error = e;
        return r;
      });
  };
```

**Comment by js2me** [comment]
It should be fixed in release 2.0.0
